### PR TITLE
Persist dataSourceId across applications under new Nav change

### PR DESCRIPTION
**Pull request by riysaxen-amzn**
### Description
* This PR persists dataSourceId across applications under new Nav change and add dataSourceId in the URL

### Issues Resolved

https://github.com/user-attachments/assets/e017361e-82c6-43b9-8440-9375167803ad

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/plugin.ts [hunk 1]
```diff
@@ -91,6 +103,7 @@ export class notificationsDashboardsPlugin
         title: 'Channels',
         order: 9070,
         workspaceAvailability: WorkspaceAvailability.outsideWorkspace,
+        updater$: this.appStateUpdater,
```
Comment (amsiglan) [inline public/plugin.ts:106]: What does this do?

Comment (riysaxen-amzn) [reply public/plugin.ts:106]: it provides global data persistence as per -> https://github.com/opensearch-project/OpenSearch-Dashboards/blob/f9965cd29afec8996a0c7b7f5e6001ab41de924e/src/plugins/opensearch_dashboards_utils/docs/global_data_persistence.md?plain=1#L20

**Comment by riysaxen-amzn** [comment]
UTs passing locally:

```
 PASS  public/pages/Main/__tests__/Main.test.tsx (34.076 s)
 PASS  public/pages/Channels/__tests__/ChannelSettingsDetails.test.tsx (23.958 s)
 PASS  public/pages/CreateChannel/__tests__/EmailSettings.test.tsx (30.083 s)
 PASS  server/utils/__tests__/helper.test.ts (8.109 s)
 PASS  public/pages/Emails/__tests__/validationHelper.test.ts (10.825 s)
 PASS  public/utils/__tests__/helper.test.ts (10.231 s)
 PASS  public/pages/CreateChannel/__tests__/validationHelper.test.ts (10.832 s)
 PASS  public/pages/Emails/__tests__/helper.test.ts (9.279 s)
 PASS  public/pages/CreateChannel/__tests__/helper.test.ts (6.319 s)
 PASS  public/components/__tests__/ContentPanelActions.test.tsx (20.771 s)
 PASS  public/components/__tests__/Modal.test.tsx (22.695 s)
 PASS  public/pages/CreateChannel/__tests__/WebhookHeaders.test.tsx (22.783 s)
 PASS  public/components/__tests__/ContentPanel.test.tsx (20.366 s)
 PASS  public/pages/Channels/__tests__/DetailsListModal.test.tsx (19.872 s)
 PASS  public/pages/Emails/__tests__/EmailGroups.test.tsx (20.635 s)
A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --detectOpenHandles to find leaks. Active timers can also cause this, ensure that .unref() was called on them.

Test Suites: 45 passed, 45 total
Tests:       3 skipped, 146 passed, 149 total
Snapshots:   83 passed, 83 total
Time:        137.11 s


```

**Comment by riysaxen-amzn** [comment]
need to check with osd team why installation of notification plugin in osd is failing

**Review by amsiglan — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Comment by riysaxen-amzn** [comment]
e2e cypress tests are failing because Alerting backend is not backported fully it seems ->

```
 Task :alerting:compileJava
/home/runner/work/alerting-dashboards-plugin/alerting-dashboards-plugin/alerting/alerting/src/main/java/org/opensearch/percolator/PercolateQueryBuilderExt.java:116: error: PercolateQueryBuilderExt is not abstract and does not override abstract method fieldName() in AbstractQueryBuilder
public class PercolateQueryBuilderExt extends AbstractQueryBuilder<PercolateQueryBuilderExt> {
```

**Pull request merged**